django/src/rdwatch/tasks/__init__.py

```diff
@@ -596,9 +598,27 @@ def generate_site_images_for_evaluation_run(
 @shared_task
 def generate_image_embedding(id: int):
     site_image = SiteImage.objects.get(pk=id)
+    timeout = 1200  # timeout for downloading if it takes longer than 20 minutes
+    polling_interval = 30
     try:
         logger.warning('Loading checkpoint Model')
         checkpoint = '/data/SAM/sam_vit_h_4b8939.pth'
+        if not os.path.exists(
+            checkpoint
+        ):  # somehow the initial download didn't happen so we download the model again
+            #  This may download the file twice if the user attempts
+            # to do SAM before the initial download completes
+            # this is a temporary solution until we start reporting SAM model
+            # status in the /status endpoint and reflect this
+            # in the front-end to prevent SAM before the model is downloaded.
+            download_sam_model_if_not_exists()
```

Comment:
I tested this locally, and it looks like celery will not pull any tasks off the queue until the `worker_ready` signal handler has finished executing. So, this file should always be available during this task, so I don't think this logic is needed. 
Did you observe the worker kicking off a task before the signal handler finished?

Reply:
I never directly observed it.  So I'll remove this, thanks for checking.